up42/tools.py

```diff
@@ -224,6 +223,88 @@ def get_block_details(self, block_id: str, as_dataframe: bool = False) -> dict:
         else:
             return details_json
 
+    def get_credits_balance(self) -> dict:
+        """
+        Display the overall credits available in your account.
+
+        Returns:
+            A dict with the balance of credits available in your account.
+        """
+        if not hasattr(self, "auth"):
+            raise Exception(
+                "Requires authentication with UP42, use up42.authenticate()!"
+            )
+        endpoint_url = f"{self.auth._endpoint()}/accounts/me/credits/balance"
+        response_json = self.auth._request(request_type="GET", url=endpoint_url)
+        details_json = response_json["data"]
+        return details_json
+
+    def get_credits_history(
+        self,
+        start_date: Optional[Union[str, datetime]] = None,
+        end_date: Optional[Union[str, datetime]] = None,
+    ) -> Dict[str, Union[str, int, Dict]]:
+        """
+        Display the overall credits history consumed in your account.
+        The consumption history will be returned for all workspace_ids on your
+        account.
+        Args:
+            start_date: The start date for the credit consumption search e.g.
+            2021-12-01. Default start_date None uses 2000-01-01.
+            end_date: The end date for the credit consumption search e.g.
+            2021-12-31. Default end_date None uses current date.
+
+        Returns:
+            A dict with the information of the credit consumption records for
+            all the users linked by the account_id. The result match the endpoint
```

Comment:
"The result match the endpoint response with all records on the content key."
Not sure what this means, possible to clarify or remove?